packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx

```diff
@@ -220,12 +235,25 @@ function DashboardLayout(props: DashboardLayoutProps) {
               <Box sx={{ mr: 1 }}>
                 <LogoContainer>{branding?.logo ?? <ToolpadLogo size={40} />}</LogoContainer>
               </Box>
-              <Typography variant="h6" sx={{ color: (theme) => theme.palette.primary.main }}>
+              <Typography
+                variant="h6"
+                sx={{ color: (theme) => (theme.vars ?? theme).palette.primary.main }}
+              >
                 {branding?.title ?? 'Toolpad'}
               </Typography>
             </Stack>
           </a>
           <Box sx={{ flexGrow: 1 }} />
+          {hasMounted && isDualTheme ? (
```

Comment:
If possible I'd like to render as much as possible during SSR. We can basicaly show everything except the icon. (Technically I guess under the css vars theme we can render both icons and use css vars to show one or the other)


Instead of turning on/off the whole thing during ssr, can we do something equivalent to
```tsx
{isDualTheme ? (
  <Tooltip title={isSsr ? null : `${paletteMode === 'dark' ? 'Light' : 'Dark'} mode`} enterDelay={500}>
    <IconButton
      aria-label={isSssr ? 'Change theme' : `Switch to ${paletteMode === 'dark' ? 'light' : 'dark'} mode`}
      onClick={toggleMode}
    >
      {isSsr ? <EmptyIcon /> : paletteMode === 'dark' ? <LightModeIcon /> : <DarkModeIcon />}
    </IconButton>
  </Tooltip>
) : null}
```

This will avoid a horizontal layout shift

Reply:
I followed the documented examples for a theme switcher in CSS vars and don't remember if there was an extra reason for doing this, will look into it, if it's possible to server-side render the switcher it sounds good.

Reply:
As far as I know the only reason is hydration mismatches. But not all usage of our component will be hydrated, that's why we should use `useSsr`. And even when hydrated, we should only stabilize what causes the hydration error. i.e. the icon.

Reply:
Not sure if it's possible to make the correct icon show during server render as the icons are rendered by Javascript, as opposed to the theme which is just CSS.
I went with a solution kinda like you suggested, it seems great!